openedx_external_enrollments/edxapp_wrapper/backends/course_settings_j_v1.py

```diff
@@ -0,0 +1,79 @@
+"""Backend for course settings module."""
+import logging
+import time
+
+from cms.djangoapps.models.settings.course_metadata import CourseMetadata
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx_external_enrollments.models import OtherCourseSettings
+from xmodule.modulestore.django import modulestore
+
+LOG = logging.getLogger(__name__)
+
+
+def update_course_settings(*args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Updates external_course_id and external_platform from a course overview.
+    This function is called from edx-platform once advance course settings is successfully saved.
+    """
+
+    try:
+        course_overview = CourseOverview.get_from_id(kwargs.get('course_key'))
+        other_course_settings = kwargs.get('other_course_settings')
+
+        OtherCourseSettings.objects.update_or_create(
+            course_overview=course_overview,
+            defaults={
+                'external_course_id': other_course_settings.get('external_course_run_id', ''),
+                'external_platform': other_course_settings.get('external_platform_target', ''),
+            }
+        )
+    except Exception as error:  # pylint: disable=broad-except
+        LOG.error('Failed to update course_settings in the backend. Reason: %s', str(error))
+
+
+def migrate_course_settings(*args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Migrates all course settings to OtherCourseSettings table in edxapp database.
+    delay_seconds -> sleep time in seconds.
+    group_length -> number of courses to be migrated by groups.
+    """
+
+    module_store = modulestore()
+    courses = module_store.get_courses()
+    migrated_courses = []
+
+    try:
+        group_counter = 0
+
+        for course in courses:
+            group_counter += 1
+            course_overview = CourseOverview.get_from_id(course.id)
+            advanced_dict = CourseMetadata.fetch(course)
+            other_course_settings_key = advanced_dict.get('other_course_settings')
+            other_course_settings = other_course_settings_key.get('value') if other_course_settings_key else {}
+
+            # Only save or update courses that have other_course_settings configurations.
+            if other_course_settings and other_course_settings.get('external_course_run_id'):
+                migrated_courses.append(str(course_overview.id))
+                OtherCourseSettings.objects.update_or_create(
+                    course_overview=course_overview,
+                    defaults={
+                        'external_course_id': other_course_settings.get('external_course_run_id', ''),
+                        'external_platform': other_course_settings.get('external_platform_target', ''),
+                    }
+                )
+
+            # Sleeps in every group_length reach to avoid database from crashing.
+            if group_counter % kwargs.get('group_length') == 0:
```

Comment:
```suggestion
            if group_counter == kwargs.get('group_length', 0):
```
Please add a default value to group length. @ivanvgh

Reply:
This argument will always be present since it is required, and if we put '==' it will only sleep in the first group, that is why the group_counter should be multiple of group_length to sleep in every group_length touched.

Reply:
Great, I didn't see it in that way.